### PR TITLE
Handled subcats of Category:Draft articles 

### PR DESCRIPTION
**Pull request by satcasm**
Issue reference: #131

### CODE REVIEW

File: src/modules/submissions.js [hunk 1]
```diff
@@ -483,7 +483,25 @@
 		return this.text;
 	};
 
-	AFCH.Text.prototype.cleanUp = function ( isAccept ) {
+	function getDraftArticleSubcatsInner( cat ) {
+		return new mw.Api().get( { action: 'query', list: 'categorymembers', cmtitle: 'Category:' + cat, cmtype: 'subcat' } ).then( function ( data ) {
+			return $.when.apply( $, data.query.categorymembers.map( function ( e ) {
+				var txt = e.title.replace( /Category:(.*?\s*)/gi, '$1' );
+				return getDraftArticleSubcatsInner( txt ); // changed it to return apiCall here
```
Comment (enterprisey) [inline src/modules/submissions.js:490]: nitpick: delete the comment on this line ("changed it...") and two lines down ("starting with cat here..."); below in getDraftArticleSubcats there also seem to be left-over comments that should also be removed

File: src/modules/submissions.js [hunk 2]
```diff
@@ -533,9 +551,14 @@
...
-			text = text.replace( /\[\[Category:/gi, '[[:Category:' );
+			text = text.replace( /\[\[:?Category:(.*?\s*)\]\]/gi, function call( match, category ) {
+				if ( draftArticleSubcats.includes( category ) ) {
+					return '[[Category:' + category + ']]';
+				} else {
+					return '[[:Category:' + category + ']]';
+				}
+			} );
 		}
-
```
Comment (enterprisey) [inline src/modules/submissions.js:538]: nitpick: here and further down, there are a bunch of removed lines; please add them back in so the diff is smaller (ideally, the diff is as small as possible)

File: src/modules/submissions.js [hunk 3]
```diff
@@ -2027,162 +2072,166 @@
...
-				// ARTICLE
-				// -------
-
-				newText.removeAfcTemplates();
-
-				newText.updateCategories( data.newCategories );
+			{ movetalk: true } ), // Also move associated talk page if exists (e.g. `Draft_talk:`)
+		getDraftArticleSubcats( 'Draft articles' )
+		).done( function ( moveData, draftArticleSubcats ) {
+			// draftArticleSubcats.forEach( function ( e ) {
```
Comment (enterprisey) [inline src/modules/submissions.js:2080]: commented-out code here should be removed

File: src/modules/submissions.js [hunk 4]
```diff
@@ -2027,162 +2072,166 @@
 
 	function handleAccept( data ) {
 		var newText = data.afchText;
-
-		AFCH.actions.movePage( afchPage.rawTitle, data.newTitle,
+		$.when( AFCH.actions.movePage( afchPage.rawTitle, data.newTitle,
 			'Publishing accepted [[Wikipedia:Articles for creation|Articles for creation]] submission',
-			{ movetalk: true } ) // Also move associated talk page if exists (e.g. `Draft_talk:`)
-			.done( function ( moveData ) {
```
Comment (enterprisey) [inline src/modules/submissions.js:2034]: this whole block (the callback) seems to have had one level of indentation removed; please indent it by one level again, to reduce the size of the diff

**Review by enterprisey — CHANGES_REQUESTED** [review]
just a few changes for now, will take a look again tomorrow

File: src/modules/submissions.js (hunk 2 above)
Comment (satcasm) [reply src/modules/submissions.js:538]: Did you mean the blank lines which were removed?

File: src/modules/submissions.js (hunk 4 above)
Comment (satcasm) [reply src/modules/submissions.js:2034]: But if I indent it by one level I am getting an error in grunt build.

File: src/modules/submissions.js (hunk 2 above)
Comment (enterprisey) [reply src/modules/submissions.js:538]: Yes, the blank lines

File: src/modules/submissions.js (hunk 4 above)
Comment (enterprisey) [reply src/modules/submissions.js:2034]: Which error?

Comment (satcasm) [reply src/modules/submissions.js:2034]: Tab indentation error that 5 tabs are required instead of 4

File: src/modules/submissions.js [hunk 5]
```diff
@@ -706,6 +734,30 @@
 		}
 	} );
 
+	function getDraftArticleSubcats( cat ) {
```
Comment (enterprisey) [inline src/modules/submissions.js:737]: Can this function be moved up next to the getDraftArticleSubcatsInner function?

File: src/modules/submissions.js [hunk 6]
```diff
@@ -706,6 +734,30 @@
...
+			lsKey = 'afch-' + AFCH.consts.version + 'draft-article-subcats';
+		if ( window.localStorage && window.localStorage[ lsKey ] ) {
+			articleSubcats = JSON.parse( window.localStorage[ lsKey ] );
+			return articleSubcats;
+		} else {
+			return getDraftArticleSubcatsInner( cat ).then( function ( articleSubcats ) { // there has to be a return here so that the function will return something
+				// If possible, cache the articleSubcats data!
+				if ( window.localStorage ) {
+					try {
+						window.localStorage.removeItem( lsKey );
```
Comment (enterprisey) [inline src/modules/submissions.js:749]: this line was only necessary during testing, so it can be removed now

**Review by enterprisey — CHANGES_REQUESTED** [review]
just one more then I think it's good

File: src/modules/submissions.js (hunk 6 above)
Comment (satcasm) [reply src/modules/submissions.js:749]: Are you talkin about ``window.localStorage.removeItem( lsKey );`` ?

**Comment by enterprisey** [comment]
I don't have the time to fix this (which was my original plan), so I think we'll have to close this and let someone else do the feature.

**Pull request closed without merging**